src/extension/react_table.js

```diff
@@ -37,41 +52,74 @@ function Table(props) {
     headerGroups,
     rows,
     prepareRow,
-  } = useTable({ columns, data })
+    allColumns,
+    getToggleHideAllColumnsProps,
+  } = useTable({ columns, data });
+
+  const toggleExpand = () => {
+    setCollapse(!collapse)
+  };
 
   return (
-    <div className="recordTable">
-      <table {...getTableProps()}>
-        <thead>
-          {headerGroups.map(headerGroup => (
-            <tr {...headerGroup.getHeaderGroupProps()}>
-              {headerGroup.headers.map(column => (
-                <th {...column.getHeaderProps()}>
-                  {column.render('Header')}
-                </th>
-              ))}
-            </tr>
-          ))}
-        </thead>
-        <tbody {...getTableBodyProps()}>
-          {rows.map(row => {
-            prepareRow(row)
-            return (
-              <tr {...row.getRowProps()}>
-                {row.cells.map(cell => {
-                  return (
-                    <td
-                      {...cell.getCellProps()}>
-                      {cell.render('Cell')}
-                    </td>
-                  )
-                })}
+    <>
+      <div className="field_select_group">
+        <span onClick={toggleExpand.bind(this)}
+          className={cx("glyphicon fieldsIcon", {
+            "glyphicon-collapse-down": !collapse,
+            "glyphicon-collapse-up": collapse
+          })} />
+        {collapse ?
+          <div className="popBoxField">
+            <div >
+              <label>
+                <IndeterminateCheckbox {...getToggleHideAllColumnsProps()} /> Toggle All
+              </label>
+            </div>
+            {allColumns.map(column => (
+              <div key={column.id}>
+                <label>
+                  <input type="checkbox" {...column.getToggleHiddenProps()} />{' '}
+                  {column.Header}
+                </label>
+              </div>
+            ))}
+          </div>
+          :
+          null
```

Comment:
We could write the popup window in a react component wrapper if we want.

Reply:
If we end up using it more than once, we probably should.

Reply:
But I think since it only shows up once, it's ok as-is.

Reply:
For right now, we only use it once.